halotools/empirical_models/factories/model_factory_template.py

```diff
@@ -276,12 +276,12 @@ def update_param_dict_decorator(self, component_model, func_name):
         :ref:`param_dict_mechanism`
         """
 
-        def decorated_func(*args, **kwargs):
+        def decorated_func(*args, __param_dict__=self.param_dict, **kwargs):
```

Comment:
@rainwoodman - this line seems to be causing problems for a few builds. Any idea why? See https://travis-ci.org/astropy/halotools/jobs/399786150.

Reply:
It's incompatible with Python 2.7. I pushed a fix.